### PR TITLE
[RW-727] Drupal 10 upgrade

### PR DESCRIPTION
**Pull request by orakili**
Refs: RW-727

This upgrades RW to D10 and PHP 8.2 and include the local stack setup + some fixes.

## Tests

1. Set up local stack
   1. Copy `local/.env.example` to `local/.env`, change the `SITE_DOMAIN` to your liking or keep the default `rwint-local.test`, set the `MAPBOX_TOKEN` (if you have it)
   2. Copy `local/shared/settings/settings.site.php.example` to `local/shared/settings/settings.site.php`, adjust the relevant settings  (notably set the `reliefweb_api.settings.api_url`, `reliefweb_api.settings.api_url_external` and `reliefweb_api.settings.elasticsearch` to something appropriate (for example with what is described in https://github.com/UN-OCHA/rwint-api/pull/152)
   3. Run `./local/install.sh -m -c -i` to create an instance of the site
   4. Import a DB dump (recent light dump for example) with `./local/exec.sh -u appuser -T drupal drush sqlc < path-to-db-dump.sql`
   5. Run `./local/install.sh -d` to install and enable the dev dependencies
2. Index content with `./local/exec.sh -u appuser drupal drush rapi-i --tag=local --replace --alias all`
3. Click around (maps may not display as maps if no valid mapbox token from `1.i`, that's fine), create some content etc. to validate it's working fine.

### CODE REVIEW

**Comment by orakili** [comment]
@lazysoundsystem I already did several rounds of thorough testing (shame on me if you find some issues after saying that) and it's been deployed to the `demo` site so the review is mainly to get approval to merge in `develop` and also, of course, report any glaring issue.

File: Makefile [hunk 1]
```diff
@@ -1,6 +1,8 @@
 # Variables. Yes.
 DOCKER=docker
 DOCKER_BUILDKIT=0
+IMAGE_NAME=rwint9-site
```
Comment (lazysoundsystem) [inline Makefile:4]: A good time to rename the image! Could be just `rwint-site` to match the value in `local/.env.example` (and time to rename the repo too, but that's probably worth its own ticket.)

**Review by lazysoundsystem — APPROVED** [review]
Looking good! Tested locally and all works well.

If there are any lessons from the change to CKEditor that might apply to the other repos, would be good to note them in https://humanitarian.atlassian.net/browse/OPS-9268

File: Makefile (hunk 1 above)
Comment (orakili) [reply Makefile:4]: Yes, I'd also like to rename it but also already have a `rwint-site` repo and that will probably imply several changes on the config used on the server. Let's make an OPS ticket and check with Cafuego.

File: patches/.gitkeep
[the remark this answers is not in the document]
Comment (orakili) [reply patches/.gitkeep:1]: Thanks, I didn't see it on my computer as the filesystem is not case sensitive... Removed.

File: composer.json [hunk 2]
```diff
@@ -137,16 +142,38 @@
...
+                "type:drupal-custom-module"
+            ],
+            "html/themes/custom/{$name}": [
+                "type:drupal-custom-theme"
+            ],
+            "drush/Commands/{$name}": [
+                "type:drupal-drush"
+            ]
         },
+        "preserve-paths": [
```
[the remark this answers is not in the document]
Comment (orakili) [reply composer.json]: Thanks. Removed.

File: composer.json [hunk 3]
```diff
@@ -20,68 +20,71 @@
         }
     ],
     "require": {
-        "php": "8.0.*",
-        "composer/installers": "^1.10",
+        "php": ">=8.1",
```
[the remark this answers is not in the document]
Comment (orakili) [reply composer.json]: Thanks. Bumped.

**Comment by lazysoundsystem** [comment]
@orakili - just noting the Drush patch was only in the `patches` directory. Now it's still in composer.patches.json https://github.com/UN-OCHA/rwint9-site/blob/38d48df358a01b7c48ece91a2ab2d2ae28d06f72/composer.patches.json#L21, but isn't in the `PATCHES` directory - all the other patches are present.

**Pull request merged**